### ISSUE TITLE
Version 0.6 does not compile on 32 bit Linux (compiler crash)

### ISSUE THREAD

**Issue by peti** [ku-fpg/blank-canvas#76]
Is this a known issue?

~~~~
[  190s] [ 4 of 17] Compiling Graphics.Blank.JavaScript ( Graphics/Blank/JavaScript.hs, dist/build/Graphics/Blank/JavaScript.p_o )
[  198s] ghc: panic! (the 'impossible' happened)
[  198s]   (GHC version 7.10.3 for i386-unknown-linux):
[  198s] 	Simplifier ticks exhausted
[  198s]   When trying UnfoldingDone eta_aoW1
[  198s]   To increase the limit, use -fsimpl-tick-factor=N (default 100)
[  198s]   If you need to do this, let GHC HQ know, and what factor you needed
[  198s]   To see detailed counts use -ddump-simpl-stats
[  198s]   Total ticks: 8
~~~~

**Comment by andygill** [comment]
I've never seen this. @roboguy13 can you try reproduce this?

**Comment by peti** [comment]
We're building with --enable-shared. Maybe that makes a difference?

Anyway, the build error does away if we set -fsimpl-tick-factor=200.

**Comment by roboguy13** [comment]
@peti Hmm, I haven't fully tried it out yet but are you sure that there isn't something in a config file that is setting the simpl ticks to a low value? Also, the `Total ticks: 8` part of the output looks particularly unusual to me because usually the `Total ticks` is significantly higher (even if I set `-fsimpl-tick-factor=8` it's usually more than ten thousand on my machine).

**Comment by peti** [comment]
> are you sure that there isn't something in a global config file somewhere that is setting the simpl ticks to a low value?

Yes, I am sure. These builds run in a clean VM that's set up only for the purposes of compiling this package.

> What happens if you try to explicitly give it -fsimpl-tick-factor=100?

I don't know about "100", but setting "200" fixes the issue.

**Comment by roboguy13** [comment]
I was able to reproduce this exactly on a 32-bit VM running a server installation of Ubuntu when installing with profiling enabled (it failed on the exact same file with the same `Total ticks: 8` message when building the profiling part of the package). It looks like it is partly a result of a line the `blank-canvas.cabal` file that sets `-fsimpl-tick-factor= 100000` when profiling is enabled. If I halve that and change it to set `-fsimpl-tick-factor=50000` when profiling is enabled, the problem disappears for me.

This seems like an issue in GHC 7.10.3 (since it is only saying, and consistently saying, the total ticks is 8 in the error), but this could be a way to workaround whatever the underlying issue is.

@andygill Is there a reason that the tick-factor needs to be set that high when profiling is enabled or would it be reasonable to cut it in half?

**Comment by andygill** [comment]
You're proposed solution sounds reasonable to me.

**Comment by roboguy13** [comment]
Fixed in 646a01061d914ecc459ca0c9cb13ae7060358d7a
